src/profiling.md

```diff
@@ -12,5 +12,84 @@ Depending on what you're trying to measure, there are several different approach
     See [their docs](https://github.com/rust-lang/measureme/blob/master/summarize/Readme.md) for more information.
     
 - If you want function level performance data or even just more details than the above approaches:
-  - Consider using a native code profiler such as [perf](profiling/with_perf.html).
+  - Consider using a native code profiler such as [perf](profiling/with_perf.html) 
+  - or [tracy](https://github.com/nagisa/rust_tracy_client) for a nanosecond-precision, 
+    full-featured graphical interface.
 
+- If you want a nice visual representation of the compile times of your crate graph, 
+  you can use [cargo's `-Ztimings` flag](https://doc.rust-lang.org/cargo/reference/unstable.html#timings), 
+  eg. `cargo -Ztimings build`.
+  You can use this flag on the compiler itself with `CARGOFLAGS="-Ztimings" ./x.py build`
+  
+## Optimizing rustc's self-compile-times with cargo-llvm-lines
+
+Using [cargo-llvm-lines](https://github.com/dtolnay/cargo-llvm-lines) you can count the 
+number of lines of LLVM IR across all instantiations of a generic function.
+Since most of the time compiling rustc is spent in LLVM, the idea is that by
+reducing the amount of code passed to LLVM, compiling rustc gets faster.
+
+Example usage:
+```
+cargo install cargo-llvm-lines
+# On a normal crate you could now run `cargo llvm-lines`, but x.py isn't normal :P
+
+# Do a clean before every run, to not mix in the results from previous runs.
+./x.py clean
+RUSTFLAGS="--emit=llvm-ir" ./x.py build --stage 0 compiler/rustc
```

Comment:
Hmm, does this work? I thought you needed to `export RUSTFLAGS` for it to be passed to child processes, but maybe bootstrap handles it specially.

Reply:
I thought `ENV_VAR='abc' foo` was the same as

```sh
$ export ENV_VAR='abc'
$ foo
```

except that it was only for that command. But maybe I don't understand my shell ;)

Reply:
Export passes it to processed that `foo` spawns, using it inline does not. Your example is only the same if foo does not spawn child processes, but bootstrap launches lots of them.

Reply:
Or at least, that seems to have been the behavior when I tried it out. Maybe I'm just misunderstanding how bootstrap works.

Reply:
I always do it this way, so it works.  I guess bootstrap handles it, like it does with `CARGOFLAGS`.